treeherder/perf/alerts.py

```diff
@@ -83,6 +96,29 @@ def generate_new_alerts_in_series(signature):
                     prev_value, new_value, signature.lower_is_better
                 )
 
+                noise_profile = "N/A"
+                try:
+                    # Gather all data up to the current data point that
+                    # shows the regression and obtain a noise profile on it.
+                    # This helps us to ignore this alert and others in the
+                    # calculation that could influence the profile.
+                    noise_data = []
+                    for point in analyzed_series:
+                        if point == cur:
+                            break
+                        noise_data.append(geomean(point.values))
+
+                    noise_profile, _ = deviance(noise_data)
+
+                    if not isinstance(noise_profile, str):
+                        raise Exception(
+                            "Expecting a string as a " f"noise profile, got: {type(noise_profile)}"
+                        )
+                except Exception:
+                    # Fail without breaking the alert computation
+                    logger.warning("Failed to obtain a noise profile.")
+                    traceback.print_exc()
```

Comment:
You could add new relic reporting here too or instead. There should be examples in the codebase.